src/dlls/profattach/profattach.cpp

```diff
@@ -0,0 +1,115 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+// 
+// ProfAttach.cpp
+// 
+
+// 
+// Definitions of functions that help with attaching and detaching profilers
+// This is the main source of profattach.dll, which is used for profiler attach purposes 
+// 
+
+// ======================================================================================
+
+#ifndef FEATURE_PAL
+#include <Windows.h>
+#endif
+
+#include <winwrap.h>
+#include <utilcode.h>
+#include <log.h>
+#include <tlhelp32.h>
+#include <cor.h>
+#include <sstring.h>
+#include "profattach.h"
+#include <log.h>
+#include <ex.h>
+#include <getproductversionnumber.h>
+
+#include "../../src/vm/profattach.h"
+#include "../../src/vm/profattachclient.h"
+#include "../../src/pal/prebuilt/inc/metahost.h"
+
+#ifndef FEATURE_PAL
+#define PSAPI_VERSION 2
+#include <psapi.h>
+#endif
+
+
+#ifdef FEATURE_PAL
+#define INITIALIZE_SHIM { if (PAL_InitializeDLL() != 0) return E_FAIL; }
+#else
+#define INITIALIZE_SHIM
+#endif
+
+// Contract for public APIs. These must be NOTHROW.
+#define PUBLIC_CONTRACT \
+    INITIALIZE_SHIM \
+    CONTRACTL \
+    { \
+        NOTHROW; \
+    } \
+    CONTRACTL_END;
+
+// Function exported by coreclr to get CLRProfiling
+typedef HRESULT(STDAPICALLTYPE * fpICLRProfilingGetClassObject)(
+    REFCLSID rclsid,
+    REFIID riid,
+    LPVOID * ppv);
+
+bool
+CreateCLRProfiling(
```

Comment:
I feel embarrassed saying this, partly because its late and partly because I'm pretty sure it was me who suggested having profattach dll in the first place ... but looking at it I'm having trouble remembering why I thought we should have a 2nd dll. I think I was imagining we were saving code size in CoreCLR, but given how little this dll is doing (load CoreCLR, call ICLRProfilingGetClassObject, call CreateInstance) perhaps we should just put CreateCLRProfiling as an export of CoreCLR and eliminate the 2nd dll. Do you recall a reason we needed it that I am now forgetting? Just in case let me send an email and ask a few other folks.

Reply:
We are primarily optimizing the total .NET Core Runtime package size, not size of coreclr.dll/.so.

Also, profattach depends on libcoreclr.so on Unix so it is not really a small independent .dll there...